src/resolution/graph.rs

```diff
@@ -1271,6 +1284,25 @@ impl<'a, TNpmRegistryApi: NpmRegistryApi>
       NpmDependencyEntryKind::Peer | NpmDependencyEntryKind::OptionalPeer
     ));
 
+    if peer_dep.kind.is_optional()
```

Comment:
I flipped this condition when cleaning up. Will fix (was just testing in integration)